AAT_gae/src/main/java/com/aat/restlet/AttendanceResource.java

```diff
@@ -88,32 +88,41 @@ public String update(Representation entity) {
 		Long groupId = Long.parseLong(getAttribute(Constants.groupId), 10);
 		String flagMode = ResourceUtil.getParam(params, Constants.flagMode, true);
 		String dateWeek = ResourceUtil.getParam(params, Constants.dateWeek, true);
+		String token = ResourceUtil.getParam(params, Constants.token, true);
+		
 		Date date;
 		AttendanceRecord attendance = retrieveAttendanceRecord(studentId,groupId);
-		List <Date> dates = null;
-		try {
-			date = new SimpleDateFormat("dd-MM-yyyy").parse(dateWeek);
-			
-			if (flagMode.equals("A")){
-					dates = attendance.getAttendance();
-			}
-			else if (flagMode.equals("P")){
-					dates = attendance.getPresentation();
-			}
-			else{
-				throw new ResourceException(409, "Conflict", "Invalid flag.", null);
-			}
+		String storedToken = attendance.getAttendaceToken().get(dateWeek);
+		
+		if (storedToken!=null && storedToken.equals(token)){
+			List <Date> dates = null;
+			try {
+				date = new SimpleDateFormat("dd-MM-yyyy").parse(dateWeek);
+				
+				if (flagMode.equals("A")){
+						dates = attendance.getAttendance();
+				}
+				else if (flagMode.equals("P")){
+						dates = attendance.getPresentation();
+				}
+				else{
+					throw new ResourceException(409, "Conflict", "Invalid flag.", null);
+				}
+				
+				if (!dates.contains(date)){
+					dates.add(date);
+				}else{
+					throw new ResourceException(409, "Conflict", "The student already has an attendance record for this week", null);
```

Comment:
Question, does it really matter if they have already marked attendance for this date?